source/pool/contracts/Pool.sol

```diff
@@ -0,0 +1,161 @@
+/*
+  Copyright 2020 Swap Holdings Ltd.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/utils/Pausable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/cryptography/MerkleProof.sol";
+
+/**
+ * @title Pool: Claim Tokens Based on a Pricing Function
+ */
+contract Pool is Ownable, Pausable {
+  using SafeMath for uint256;
+
+  // Higher the scale, lower the output for a claim
+  uint256 public immutable scale;
```

Comment:
I would use uppercase for SCALE and MAX so that it is easier to understand that they are constants

Reply:
We might want to make these mutable so admins can adjust if conditions change. Will take a look.